packages/earljs/src/validators/dataStructures.ts

```diff
@@ -0,0 +1,45 @@
+import { Control } from '../Control'
+import { ArrayWithMatcher, ContainerWithMatcher } from '../matchers'
+import { ArrayOfLengthMatcher } from '../matchers/ArrayOfLength'
+import { ObjectWithMatcher } from '../matchers/ObjectWith'
+import { formatValue } from './common'
+
+export function toBeAContainerWith(control: Control<any>, expectedItems: any[]) {
+  const m = new ContainerWithMatcher(expectedItems)
```

Comment:
Doing this actually loses information. There are two conditions in which this validator fails:
1. Values is not a container
2. Value does not include expected items
Using a matcher gives you boolean information and so the user stays uninformed as to the nature of the problem

Reply:
As an additional remark, we can consider runtime type checking in more matchers, even if we know the type. This will help both pure JS users as well as TS users that don't have strict types or do unsafe casting.

Reply:
I agree with your remark that perhaps `5 does not contain ["a"]` is not a perfect error message but I don't consider it to be a deal breaker. 

I am strongly against doing anything more in validators. Now that I think about it - all these additional checks should be part of the matcher itself and it should be able to pass it to validators. This way we get better error messages not only for validators but for (nested) matchers too. I am gonna create a separate issue for that.